app/models/hostgroup.rb

```diff
@@ -323,4 +324,14 @@ def validate_subnet_types
     errors.add(:subnet, _("must be of type Subnet::Ipv4.")) if subnet.present? && subnet.type != 'Subnet::Ipv4'
     errors.add(:subnet6, _("must be of type Subnet::Ipv6.")) if subnet6.present? && subnet6.type != 'Subnet::Ipv6'
   end
+
+  def validate_compute_profile
+    return if compute_profile_id.nil?
+    compute_profile = ComputeProfile.find(compute_profile_id)
+    errors.add(:compute_profile, _('is not valid.')) unless valid_compute_profiles.include? compute_profile
+  end
+
+  def valid_compute_profiles
+    ComputeProfile.authorized(:view_compute_profiles).visibles.to_a
```

Comment:
This will load all compute profiles to memory, and you also perform another query above to pull out the current compute profile. Another approach can be something like:
```suggestion
    ComputeProfile.authorized(:view_compute_profiles).visibles.where(id: compute_profile_id).any?
```